doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,712 @@
+    Function classes are always {term}`symbolic`, meaning that they typically
+    remain {term}`unevaluated` when passed a {term}`symbol`, like `f(x)`. Not
+    every symbolic {term}`expression` class is a `Function` subclass, for
+    example, {term}`core` classes like `Add` and `Mul` are not `Function`
+    subclasses.
+
+    `Function` may also be used to create an {term}`undefined function` by
+    passing it a string name for the function, like `Function('f')`.
+
+    Not every function in SymPy is a symbolic `Function` class; some are just
+    Python functions which always return a value. For example, most
+    simplification functions like {term}`simplify() <simplification>` cannot
+    be represented symbolically.
+
+
+Immutable
+
+    In Python, objects are *immutable* if they can not be modified in-place.
+    In order to change an immutable object, a new object must be created. In
+    SymPy, all objects are immutable. This means that all functions that
+    operate on {term}`expressions <expression>` will return a new expression
+    and leave the original unchanged. Performing an operation on an expression
+    will never change other objects or expressions that reference that
+    expression. This also means that any two objects that are {term}`equal
+    <structural equality>` are completely interchangeable and may be thought
+    of as being the same object, even if they happen to be two different
+    objects in memory. Immutability makes it easier to maintain a mental model
+    of code, because there is no hidden state. SymPy objects being immutable
+    also means that they are hashable, which allows them to be used as
+    dictionary keys.
+
+Interactive
+
+    *Interactive* usage refers to using SymPy in an interactive REPL
+    environment such as the Python prompt, {term}`isympy`,
+    [IPython](https://ipython.org/), or the [Jupyter
+    notebook](https://jupyter.org/). When using SymPy interactively, all
+    commands are typed in real time by the user and all intermediate results
+    are shown. *Interactive* use is in contrast with *programmatic* use, which
+    is where the code is written in a file which is either executed as a
+    script or is part of a larger Python library. Some SymPy idioms are only
+    recommended for interactive use and are considered anti-patterns when used
+    programmatically. For example, running `from sympy import *` is convenient
+    when using SymPy interactively, but is generally frowned upon for
+    programmatic usage, where importing names explicitly just using `import
+    sympy` is preferred.
+
+`is_*`
+
+    Attributes in SymPy that start with *`is_`* and use a *lowercase* name
+    query the given {term}`assumption <assumptions>` on that object (note:
+    there are a few properties that are an exception to this because they do
+    not use the assumptions system). For example, `x.is_integer` will query
```

Comment:
More explanation that could be linked to here:
https://docs.sympy.org/latest/guides/assumptions.html#other-is-properties

Reply:
Thanks I didn't notice we already had documentation for this.